Feature: configurable dismiss timer duration

nfuad opened the pull request:
Closes issue https://github.com/timolins/react-hot-toast/issues/74

nfuad commented:
Updated the docs too now :)

efflam commented:
I use react-hot-toast in headless mode and handle the animation with framer-motion, this option will be really useful to be able to dismiss the toast instantly and let framer-motion remove it with AnimatePresence

timolins commented:
Thanks a lot for this PR, @nfuad. Much appreciated. Also thanks to @efflam for clarifying why this is needed.

One thing I'd like to change before merging is renaming the option to `removeDelay` (as well as the docs).
There is `toast.dismiss()` which starts the timeout, and `toast.remove()` which actually removes it. I think it would be more consistent this way, as this option delays the remove, not the dismiss.

nfuad commented:
Thank you, @efflam for clarifying the use case. In my case though, I just want to show an exit animation that is less than 1 second. 
Thank you, @timolins for taking a look. Agreed that `removeDelay` is a much more fitting name so I've renamed the option now and updated the docs too. Also added an example using the option in the `<Toaster />` component in the docs. Let me know if this is good or you want me to make some other changes. :)

timolins commented:
Thanks a lot! I'll review the PR shorty.

@nfuad Because you mentioned you need this for shorter animations: The default exit animation is also just 0.4s long. You can preserve the final state of the animation by using `animation-fill-mode: forwards;`. This way the toast stays invisible and it doesn't need to be remove at an exact time.

nfuad commented:
@timolins, ah yeah, that would work as well :)

DavoCg commented:
Hello guys, any update on this PR, with our team we have the exact same usecase as @efflam and this PR would be indeed very useful ! Thanks for your work @nfuad and @timolins for this lib we use in every of our projects !

SunsetUzuki commented:
It will be merged someday? Why it's still open?

garmeeh commented:
This would be a useful addition. The current default makes it feel a little unresponsive for the user.

irf4nmohd commented:
@nfuad  I'm new to open source. Please can you provide me a setup guide I will like to contribute. When I tried in my local the src is getting bundled and after running sites, it is picking from dist.

anirbanmajumder0 commented:
is this going to me merged soon? Would be really helpful.

samepant commented:
would love to see this PR merged :)

nfuad commented:
@cquanu I'm not sure if I can see why the vercel build is failing - please feel free to take over and merge the PR.

cquanu commented:
@nfuad Sorry I'm not the maintainer - won't be able to do this.

agreen254 commented:
I was able to somewhat do this by calling `toast.dismiss()` and wrapping `toast.remove()` in a `useTimeout()`.

My close button looks like this:
```
<button
    onClick={() => {
      toast.dismiss(t.id); // this triggers the removal animation
      setTimeout(() => toast.remove(t.id), animationTime); // toast will be removed right when the animation is over
    }}
>
  Close
</button>
```

Note that this will only work if the animation duration is less than the default of 1 second. Also not sure if this works with `AnimatePresence`.

timolins commented:
Picked this PR up again – thanks, everyone, for their patience. There was a bug where the options weren't respected when they were supplied to the Toaster itself. This was caused by some side effects, which should live outside the store, so I used this PR for a small refactor that I had wanted to do for quite some time.

Merging now!
